src/utils/layout/useExpressionDataSources.ts

```diff
@@ -37,6 +73,9 @@ export function useExpressionDataSources(): ExpressionDataSources {
   const nodeTraversal = useNodeTraversalSelectorLax();
   const transposeSelector = useDataModelBindingTranspose();
 
+  const externalApiIds = useApplicationMetadata().externalApiIds ?? [];
+  const externalApis = useExternalApis(externalApiIds);
```

Comment:
Hmm, this would mean that everyone using expression data sources (everywhere expressions are resolved) will  have to re-render whenever external API data sources change. Right now, it doesn't seem like cache invalidation and re-fetching is implemented though, so this is fine for now. :raised_hands:

I vaguely recall us discussing how that could be done, and what it would affect, but maybe we concluded to delay that until later? :thinking: Correct me if I'm wrong!

Reply:
Good point, I had totally forgotten to add `staleTime`. The data will only change when the cache entry is stale or invalidated, so I setting `staleTime` for this query to 10 min or something would ensure the data to be _relatively_ fresh whilst not having to update very often. But we probably have to have to dig a bit deeper soon to find out sane defaults, or maybe if we want the user to be able to set them 🤷‍♀️ 

What'chu think?

Reply:
The way it is now, that would mean that we'd have a once-every-10min full re-run of all expressions (in apps that use external APIs). Also, that might just make things harder to debug when we don't have any other mechanisms that behave like that. (I.e. "this is not right, let me just write a message about it to....wait, it fixed itself?? I must have been seeing things!")

We've used the `mapping` property (and the now more modern and capable `queryParameters`) for these things in components before, as a mechanism to invalidate caches and re-fetch options. I think we should introduce something later when (we know the use-cases better? or maybe that's just me). And at that point, it would be natural to use a (delayed) selector instead of the full query results - but, you know, that would also mean that we'd have to jam the query result into a zustand store.. :face_with_peeking_eye: